references/detection/train.py

```diff
@@ -164,13 +164,32 @@ def get_args_parser(add_help=True):
 
     parser.add_argument("--backend", default="PIL", type=str.lower, help="PIL or tensor - case insensitive")
     parser.add_argument("--use-v2", action="store_true", help="Use V2 transforms")
+    parser.add_argument(
+        "--with-masks",
+        action="store_true",
+        help=(
+            "Whether the dataset should return masks. Only relevant when --use-v2 is passed. "
+            "True by default when using mask_rcnn."
+        ),
+    )
 
     return parser
 
 
 def main(args):
     if args.backend.lower() == "datapoint" and not args.use_v2:
         raise ValueError("Use --use-v2 if you want to use the datapoint backend.")
+    if args.dataset not in ("coco", "coco_kp"):
+        raise ValueError(f"Dataset should be coco or coco_kp, got {args.dataset}")
+    if "keypoint" in args.model and args.dataset != "coco_kp":
+        raise ValueError("Oops, if you want Keypoint detection, set --dataset coco_kp")
+    if args.dataset == "coco_kp" and args.use_v2:
+        raise ValueError("KeyPoint detection doesn't support V2 transforms yet")
+
+    if "mask" in args.model:
+        args.with_masks = True
```

Comment:
Why would someone pass this if they didn't use a mask_rcnn model? That would only slow down the training with the no benefit, right? Maybe we should just hardcode this behavior?

Reply:
They may want to use those scripts with their own models

Reply:
They can't without modifications, since we instantiate the model like

https://github.com/pytorch/vision/blob/29418e34a94e2c43f861a321265f7f21035e7b19/references/segmentation/train.py#L160-L166

However, no strong opinion if you want to keep it.